cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -625,3 +637,52 @@ function generateProductUrl({ objectID, productUrl, queryID, indexName }) {
     url.searchParams.append('indexName', indexName);
     return url.href;
 }
+
+/**
+ * Calculate the display price for an item when lazy loading pricing is disabled
+ * @param {Object} item The item object
+ * @return {number} The calculated sales price
+ */
+function calculateDisplayPrice(item) {
+
+    var promotions;
+    var calloutMsg = '';
+
+    if (algoliaData.recordModel === 'master-level' && item.variants) {
+        promotions = item.variants && item.variants.length > 0 ? item.variants[0].promotions : item.promotions;
+        variant = item.variants && item.variants.length > 0 ? item.variants[0] : item;
```

Comment:
You are already checking `item.variants` above.
```suggestion
        promotions = item.variants.length > 0 ? item.variants[0].promotions : item.promotions;
        variant = item.variants.length > 0 ? item.variants[0] : item;
```